src/cmd/singularity/cli/build.go

```diff
@@ -97,16 +96,11 @@ var BuildCmd = &cobra.Command{
 			}
 
 			if remote {
-				if authWarning != auth.WarningEmptyToken &&
-					authWarning != auth.WarningTokenToolong &&
-					authWarning != auth.WarningTokenTooShort {
-					if authToken != "" {
-						b = build.NewRemoteBuilder(args[0], def, false, remoteURL, authToken)
-					}
+				if authWarning == "" || authToken != "" {
```

Comment:
Should it ever proceed if `authWarning` is non-empty? For example, if `authWarning` is `auth.WarningTokenTooShort`? The logic here used to `Fatalf`, was this change intentional?

Reply:
the logic I'm trying to get there is, if an env var is set, without a file, the warning is going to be not-null, but AuthToken is going to exist via the env var.
> will do more checks on the token set step.

Reply:
@ArangoGutierrez - okay, in that case can we have more explicitly named vars so that it's intuitive reading the if?

Reply:
I'm still a little confused about whether `authWarning` is meant to prevent usage of `authToken` or just warn about it. If there are cases where we want to use the token when a warning is also present, I'd suggest naming those (rather than the failure cases.) Missing a warning that should have been fatal was what caused the segmentation fault in the build code.